ReactiveCocoa/Swift/Property.swift

```diff
@@ -90,16 +90,13 @@ public final class MutableProperty<Value>: MutablePropertyType {
 	public var value: Value {
 		get {
 			lock.lock()
-			let value = _value
-			lock.unlock()
-			return value
+			defer { lock.unlock() }
+
+			return _value
 		}
 
 		set {
-			lock.lock()
-			_value = newValue
-			observer.sendNext(newValue)
-			lock.unlock()
+			modify { _ in newValue }
```

Comment:
I meant that it's exactly equivalent to the setter:
```swift
property.value = newValue
property.swap(newValue)
```

Because of this redundancy, I would get rid of `swap`.